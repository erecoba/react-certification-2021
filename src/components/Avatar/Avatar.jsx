import React from 'react';
import { Container } from './Avatar.styled';
import HappyFaceIcon from '../../ui/icons/HappyFace';

const Avatar = ({ clickable }) => {
  return (
    <Container clickable={clickable}>
      <HappyFaceIcon color="white" />
    </Container>
  );
};

export default Avatar;
