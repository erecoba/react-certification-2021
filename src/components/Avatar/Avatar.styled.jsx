import styled from 'styled-components';

const Container = styled.div`
  border-radius: 50%;
  height: 40px;
  width: 40px;
  background-color: ${({ theme }) => theme.base};

  display: flex;
  justify-content: center;
  align-items: center;

  cursor: ${({ clickable }) => (clickable ? 'pointer' : 'auto')};
`;

export { Container };
