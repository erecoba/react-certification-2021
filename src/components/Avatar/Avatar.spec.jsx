import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';
import { ThemeProvider } from 'styled-components';

import Avatar from './Avatar';

describe('Avatar', () => {
  it('should render correctly inside div', () => {
    const theme = {
      colors: {
        white: '#fff',
      },
    };
    const { container } = render(
      <ThemeProvider theme={theme}>
        <Avatar />
      </ThemeProvider>
    );

    expect(container.tagName).toMatchSnapshot('div');
  });

  it('should render with pointer cursor while clickable', () => {
    const theme = {
      colors: {
        white: '#fff',
      },
    };
    const { container } = render(
      <ThemeProvider theme={theme}>
        <Avatar clickable />
      </ThemeProvider>
    );

    expect(container.firstChild).toHaveStyleRule('cursor', 'pointer');
  });
});
