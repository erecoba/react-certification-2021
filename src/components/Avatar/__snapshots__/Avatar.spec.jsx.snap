// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar should render correctly inside div: div 1`] = `"DIV"`;
